docs/proposals/disk-affinity.md

```diff
@@ -0,0 +1,161 @@
+# Proposal: Supporting disk affinity/anti-affinity feature 
+
+Last updated: 18.01.2022
+
+## Abstract
+
+This proposal contains approach for supporting disk affinity/anti-affinity feature.
+
+## Background
+
+Current lvg storageclass cannot mount physical disks to PVs with the awareness of the requesting pods. So, it's possible
+for different applications to share the same disk.
+CSI should provide a way to support:
+* mount PVs for the same pod on the same disk (disk affinity);
+* do not mount PVs on the disks that already have some PVs from other specific pods (disk antiaffinity).
+
+## Proposal
+
+Usually for affinity/antiaffinity topology feature is used. But the minimal domain, on which this feature operates, is nodes, not disks.
+So we should use a different approach for this one. Another approach, which is proposed, is to use annotations for pointing out how to use disk affinity/antiaffinity:
+affinity.drives.csi-baremetal.dell.com/type: <pod-bound> - mount PVs for the same pod on the same disk;
+antiaffinity.drives.csi-baremetal.dell.com/type: <pod-label> - specify disk antiaffinity type: currently there is only one option - by pod labels;  
+antiaffinity.drives.csi-baremetal.dell.com/labels: "${list of the labels}" - do not mount PVs on the disks that already have some PVs from other specific pods (specified by labels).
```

Comment:
What about a case, if we don't want to allocate all PVs of one pod on the same LVG disk? For example, `www-1` and `www-1` from the sts below should be on the different disks?

Reply:
Then, we should add another antiaffinity type, e.g. called by 'pod-bound', and just set it into antiaffinity type annotation:
antiaffinity.drives.csi-baremetal.dell.com/type: <pod-label|pod-bound>

Reply:
Got it